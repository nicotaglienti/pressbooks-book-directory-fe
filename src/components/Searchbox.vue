<template>
    <ais-search-box class="searchbox" v-slot="{ refine }">
        <h2 class="ais-SearchBox__text">Find a book:</h2>
        <v-text-field
                type="search"
                v-model="stringSearch"
                label="Search all book metadata"
                @input="enableFilters(refine, stringSearch)"
        >
            <v-icon slot="append">mdi-magnify</v-icon>
        </v-text-field>
    </ais-search-box>
</template>

<script>
    export default {
        name: "Searchbox",
        data() {
            return {
                stringSearch: ""
            };
        },
        methods: {
            enableFilters(refine, currentRefinement) {
                if (currentRefinement.length > 3 || currentRefinement.length === 0) {
                    refine(currentRefinement);
                }
            },
        }
    }
</script>