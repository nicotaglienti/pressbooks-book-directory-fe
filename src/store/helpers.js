function setFilters(oldFilters, allowedFilters) {
    let fs = [], ns = [], strQuery = '', f = {};
    let query = [], queryExcluded = [], numerics = {}, q =[];
    let qtyAttr = Object.keys(oldFilters).length;
    for (let attribute in oldFilters) {
        q = [];
        for (let i =0; i < oldFilters[attribute].length; i++) {
            f = oldFilters[attribute][i];
            strQuery = ':';
            if (f.exclude) {
                if (allowedFilters[attribute].type !== 'boolean') {
                    strQuery = ':-';
                }
                queryExcluded.push(f.attribute + strQuery + f.value);
            } else {
                if (f.operator !== undefined) {
                    if (numerics[f.attribute] === undefined) {
                        numerics[f.attribute] = [];
                    }
                    numerics[f.attribute].push(f);
                } else {
                    q.push(f.attribute + strQuery + f.value);
                }
            }
        }
        if (q.length > 0) {
            query.push(q);
        }
    }
    if (queryExcluded.length > 0 && query.length > 0) {
        if (qtyAttr === 1) {
            fs.push(query);
        } else {
            fs = query;
        }
        fs = fs.concat(queryExcluded);
    }
    if (query.length > 0 && queryExcluded.length === 0) {
        fs = fs.concat(query);
    }
    if (query.length === 0 && queryExcluded.length > 0) {
        fs = fs.concat(queryExcluded);
    }
    for (let attr in numerics) {
        if (numerics[attr].length > 1) {
            let v1 = numerics[attr][0].value;
            let v2 = numerics[attr][1].value;
            if (attr === 'storageSize') {
                v1 = numerics[attr][0].value * 1024 * 1024;
                v2 = numerics[attr][1].value * 1024 * 1024;
            }
            ns.push(attr + ':' + v1 + ' TO ' + v2);
        } else {
            let v3 = numerics[attr][0].value;
            if (attr === 'storageSize') {
                v3 = numerics[attr][0].value * 1024*1024;
            }
            strQuery = numerics[attr][0].operator;
            ns.push(attr + strQuery + v3);
        }
    }
    return [fs, ns.join(' AND ')];
}

export default {
    functions: {
        setFilters
    }
};