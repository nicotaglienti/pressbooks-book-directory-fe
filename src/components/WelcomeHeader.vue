<template>
  <v-container class="welcome-header">
    <h1>Welcome to Pressbooks Directory</h1>
    <p>
      This directory provides an index of <strong>{{ $store.state.stats.numberOfBooksIndexed }}</strong> books published across
      <strong>{{ $store.state.stats.numberOfNetworksIndexed }}</strong> Pressbooks networks. Learn more about how the directory
      works by <a href="https://networkmanagerguide.pressbooks.com/chapter/how-to-use-the-pressbooks-directory/">reading our guide</a>
      or taking the self-guided tour below.
    </p>
    <v-btn
      class="directory-tour"
      aria-label="Take the tour"
      @click="$store.commit('showTour')"
    >
      Take the tour
    </v-btn>
    <!-- eslint-disable vue/no-v-html -->
    <p
      v-if="additionalText"
      v-html="additionalText"
    />
    <!--eslint-enable-->
    <collections v-if="$vuetify.breakpoint.width > 700" />
    <searchbox />
  </v-container>
</template>

<script>
import Searchbox from './Searchbox';
import Collections from './additionalCards/Collections';

export default {
  name: 'WelcomeHeader',
  components: {Collections, Searchbox},
  data() {
    return {
      additionalText: (process.env.VUE_APP_HEADER_ADDITIONAL_TEXT) ? process.env.VUE_APP_HEADER_ADDITIONAL_TEXT : false,
      totalBooks: false,
      totalNetworks: false
    };
  }
};
</script>