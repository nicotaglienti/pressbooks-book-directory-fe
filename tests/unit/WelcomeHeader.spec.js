import Vue from 'vue';
import Vuetify from 'vuetify';
import InstantSearch from "vue-instantsearch";
import { store } from "@/store/index";
import WelcomeHeader from '@/components/WelcomeHeader';
import Searchbox from '@/components/Searchbox';

Vue.use(Vuetify);

import {
  shallowMount,
  createLocalVue
} from '@vue/test-utils';

const localVue = createLocalVue();
localVue.use(Vuetify);
localVue.use(InstantSearch);

describe('WelcomeHeader.vue', () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  const wrapper = shallowMount(WelcomeHeader, {
    localVue,
    vuetify,
    store: store
  });

  it('should have the Searchbox component', () => {
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.findComponent(Searchbox).exists()).toBe(true);
  });

  it('should have the "Welcome to Pressbooks Directory title"', () => {
    const title = wrapper.find('.welcome-header__content--bold-red');
    expect(wrapper.html()).toMatchSnapshot();
    expect(title.text()).toBe('Welcome to Pressbooks Directory');
  });
});