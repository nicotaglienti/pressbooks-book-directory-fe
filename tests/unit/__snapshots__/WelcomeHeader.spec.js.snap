// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeHeader.vue should have the "Welcome to Pressbooks Directory title" 1`] = `
<v-container-stub tag="div" class="welcome-header">
  <h1>Welcome to Pressbooks Directory</h1>
  <p>
    This directory provides an index of <strong>0</strong> books published across
    <strong>0</strong> Pressbooks networks. Search and filter books by keyword,
    subject matter, license, and more. See
    <a href="https://networkmanagerguide.pressbooks.com/chapter/how-to-use-the-pressbooks-directory/">our guide for
      more detailed instructions</a> on using the Pressbooks Directory to find books of interest.
  </p>
  <!---->
  <featured-books-stub></featured-books-stub>
  <collections-stub></collections-stub>
  <searchbox-stub></searchbox-stub>
</v-container-stub>
`;

exports[`WelcomeHeader.vue should have the Searchbox component 1`] = `
<v-container-stub tag="div" class="welcome-header">
  <h1>Welcome to Pressbooks Directory</h1>
  <p>
    This directory provides an index of <strong>0</strong> books published across
    <strong>0</strong> Pressbooks networks. Search and filter books by keyword,
    subject matter, license, and more. See
    <a href="https://networkmanagerguide.pressbooks.com/chapter/how-to-use-the-pressbooks-directory/">our guide for
      more detailed instructions</a> on using the Pressbooks Directory to find books of interest.
  </p>
  <!---->
  <featured-books-stub></featured-books-stub>
  <collections-stub></collections-stub>
  <searchbox-stub></searchbox-stub>
</v-container-stub>
`;
